src/io.cpp

```diff
@@ -97,16 +97,24 @@ const Catalog &CatalogRead() {
         char *tsvPath = getenv("LOST_BSC_PATH");
         catalog = BscParse(tsvPath ? tsvPath : DEFAULT_BSC_PATH);
         // perform essential narrowing
-        // remove all stars with exactly the same position as another, keeping the one with brighter magnitude
+        // remove all stars with exactly the same position as another, keeping the one with brighter
+        // magnitude
         std::sort(catalog.begin(), catalog.end(), [](const CatalogStar &a, const CatalogStar &b) {
-            return a.spatial.x < b.spatial.x;
+            if (a.spatial.x != b.spatial.x) {
+                return a.spatial.x < b.spatial.x;
+            } else if (a.spatial.y != b.spatial.y) {
+                return a.spatial.y < b.spatial.y;
+            }
+            return a.spatial.z < b.spatial.z;
         });
-        for (int i = catalog.size(); i > 0; i--) {
-            if ((catalog[i].spatial - catalog[i-1].spatial).Magnitude() < 5e-5) { // 70 stars removed at this threshold.
-                if (catalog[i].magnitude > catalog[i-1].magnitude) {
-                    catalog.erase(catalog.begin() + i);
-                } else {
+        for (int i = catalog.size() - 1; i > 0; i--) {  // [BUG]? catalog[catalog.size()] is invalid
+            if ((catalog[i].spatial - catalog[i - 1].spatial).Magnitude() < 5e-5) {  // 70 stars removed at this threshold.
+                if (catalog[i].magnitude > catalog[i - 1].magnitude) {
+                    // [BUG]? If mag of i > mag of i-1, we want to keep i, not i-1
```

Comment:
you fool, brighter magnitudes have smaller `CatalogStar.magnitude` values! (might be worth adding a comment here to avoid confusion from future contributors)

(This behavior, for `CatalogStar` objects, is different than for `Star` objects. That's because, logically, the `CatalogStar.magnitude` is supposed to correspond to what you'd find in any astronomical database, while `Star.magnitude` is just supposed to be a relative measure of magnitude, since any estimate of magnitude from a centroiding algorithm is going to be pretty rough.